src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SonarLintTracker
+{
+    class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        // We're not using an immutable list here but we cannot modify the list in any way once we've published the snapshot.
+        public readonly List<IssueMarker> IssueMarkers = new List<IssueMarker>();
+
+        public IssuesSnapshot NextSnapshot;
+
+        internal IssuesSnapshot(string filePath, int versionNumber)
+        {
+            this.filePath = filePath;
+            this.versionNumber = versionNumber;
+        }
+
+        public override int Count
+        {
```

Comment:
In case you are concerned with LOC, this and the following property could be replaced with more modern syntax:
public override int Count => this.IssueMarkers.Count;